src/Core/Services/GraphQLSchemaCreator.cs

```diff
@@ -227,12 +233,21 @@ private DocumentNode GenerateSqlGraphQLObjects(RuntimeEntities entities, Diction
                 }
             }
 
+            // For all the fields in the object which hold a foreign key reference to any referenced entity, add a foriegn key directive.
+            AddReferencingFieldDirective(entities, objectTypes);
+
             // Pass two - Add the arguments to the many-to-* relationship fields
             foreach ((string entityName, ObjectTypeDefinitionNode node) in objectTypes)
             {
                 objectTypes[entityName] = QueryBuilder.AddQueryArgumentsForRelationships(node, inputObjects);
             }
 
+            // Create ObjectTypeDefinitionNode for linking entities. These object definitions are not exposed in the schema
```

Comment:
No, these are not required. For example: consider the `book_author_link` linking table linking `book` and `author` table. I need the linking object for `book_author_link` so that I can successfully create the linking object definition for `LinkingObjectBookAuthor` which allows insertion in a nested author entity. But I would never need to have the Object definition for linking object in the schema.